specs/sharding/beacon-chain.md

```diff
@@ -103,460 +101,342 @@ The following values are (non-configurable) constants used throughout the specif
 ```
 
-### `AttestedDataCommitment`
+### `ShardedCommitmentsContainer`
 
 ```python
-class AttestedDataCommitment(Container):
-    # KZG10 commitment to the data, and length
-    commitment: DataCommitment
-    # hash_tree_root of the ShardBlobHeader (stored so that attestations can be checked against it)
-    root: Root
-    # The proposer who included the shard-header
-    includer_index: ValidatorIndex
-```
+class ShardedCommitmentsContainer(Container):
+    sharded_commitments: List[KZGCommitment, 2 * MAX_SHARDS]
 
-### `ShardBlobBody`
+    # Aggregate degree proof for all sharded_commitments
+    degree_proof: KZGCommitment
 
-Unsigned shard data, bundled by a shard-builder.
-Unique, signing different bodies as shard proposer for the same `(slot, shard)` is slashable.
+    # The sizes of the blocks encoded in the commitments (last intermediate and all beacon blocks since)
+    included_beacon_block_sizes: List[uint64, MAX_BEACON_BLOCKS_BETWEEN_INTERMEDIATE_BLOCKS + 1]
+    
+    # Number of commitments that are for sharded data (no blocks)
+    included_sharded_data_commitments: uint64
 
-```python
-class ShardBlobBody(Container):
-    # The actual data commitment
-    commitment: DataCommitment
-    # Proof that the degree < commitment.samples_count * POINTS_PER_SAMPLE
-    degree_proof: BLSCommitment
-    # The actual data. Should match the commitment and degree proof.
-    data: List[BLSPoint, POINTS_PER_SAMPLE * MAX_SAMPLES_PER_BLOB]
-    # fee payment fields (EIP 1559 like)
-    # TODO: express in MWei instead?
-    max_priority_fee_per_sample: Gwei
-    max_fee_per_sample: Gwei
+    # Random evaluation of beacon blocks + execution payload (this helps with quick verification)
+    block_verification_kzg_proof: KZGCommitment
 ```
 
-### `ShardBlobBodySummary`
-
-Summary version of the `ShardBlobBody`, omitting the data payload, while preserving the data-commitments.
-
-The commitments are not further collapsed to a single hash, 
-to avoid an extra network roundtrip between proposer and builder, to include the header on-chain more quickly.
+### `IntermediateBlockBody`
 
 ```python
-class ShardBlobBodySummary(Container):
-    # The actual data commitment
-    commitment: DataCommitment
-    # Proof that the degree < commitment.samples_count * POINTS_PER_SAMPLE
-    degree_proof: BLSCommitment
-    # Hash-tree-root as summary of the data field
-    data_root: Root
-    # fee payment fields (EIP 1559 like)
-    # TODO: express in MWei instead?
-    max_priority_fee_per_sample: Gwei
-    max_fee_per_sample: Gwei
+class IntermediateBlockBody(phase0.BeaconBlockBody):
+    attestation: Attestation
+    execution_payload: ExecutionPayload
+    sharded_commitments_container: ShardedCommitmentsContainer
 ```
 
-### `ShardBlob`
-
-`ShardBlobBody` wrapped with the header data that is unique to the shard blob proposal.
+### `IntermediateBlockHeader`
 
 ```python
-class ShardBlob(Container):
+class IntermediateBlockHeader(Container):
```

Comment:
Good question! I guess it will also require a lot of restructuring, because all the operation processing assumes that there is a proposer. But some of that we will have to change for attestation processing anyway I guess.

Reply:
So after discussion on last call, I think it makes most sense to restructure:
 * Require block builders to be validators
 * Intermediate blocks and beacon blocks can then be the same data structure and share the same block processing